src/main/java/sg/dex/starfish/DataAsset.java

```diff
@@ -128,4 +129,25 @@ default DataAsset updateMeta(String newMetaData) {
         throw new UnsupportedOperationException("This Operation is not supported");
     }
 
+    /**
+     * This method is to include the include provenace in the asset metadata.
+     * Create default provenance metadata for publishing an asset.
+     * This create a random activity ID for tracking the provenance.
+     *
+     * @return respective data asset sub class.
+     */
+    default DataAsset includeProvenace()  {
+
+        if (null != this.getMetadata().get(Constant.PROVENANCE) ) {
+            return this;
+
+        } else {
+            Map<String, Object> metaMap = this.getMetadata();
+            String  name  =null != metaMap.get("name") ?
```

Comment:
Explain this logic.

Reply:
Reverted the changes, will raise PR separately for a separate issue.